### DISCUSSION TITLE
Iterating queryparams?

### DISCUSSION THREAD

**Discussion by jakob-lilliemarck** [poem-web/poem#444] [Q&A] [answered]
Hey,
First of all, thank you for an great api framework. The ease of use with poem-openapi really made all the difference for me when comparing with other frameworks. Great work!

I'm trying to write a poem-middleware that creates a `Filter` struct from any query-params on the request object, but I can't find a way to iterate trough all of the query-params - How do I do this? Are there any examples of iterating query-params on the request object?

**Comment by sunli829** [comment, marked as answer]
so easy:  🙂

```rust
let app = Route::new()
        .at("/hello/:name", get(hello))
        .around(|ep, request| async move {
            // parse the query params
            let query_params = request.params::<Vec<(String, String)>>()?;
            for (name, value) in query_params {
                // do something...
                println!("{}: {}", name, value);
            }
            ep.call(request).await // call the next handler
        })
        .with(Tracing);
```

**Reply by jakob-lilliemarck** [reply]
Great! Thank you :smile_cat: 
What would be the standard way to deserialize the params to their respective type? I want the querystring to deserialize as any json, i.e. `int`, `bool`, `array`, `object`, `string`. I did some attempt to use `serde_json::Json::Value` for the value-type, but it just seem to always return string.
If not, perhaps there is there a way to retrieve the querystring as is, i.e. `key=1&key2[]=1&key2[]=3`?
